File: cpp/944.delete-columns-to-make-sorted.cpp
#include <string>
#include <vector>

using std::string, std::vector;

class Solution {
public:
  int minDeletionSize(vector<string> &strs) {
    int rows = strs.size(), cols = strs.front().size();
    int removed = 0;

    for (int c = 0; c < cols; c++) {
      for (int r = 1; r < rows; r++) {
        if (strs[r - 1][c] > strs[r][c]) {
          removed++;
          break;
        }
      }
    }

    return removed;
  }
};

File: rust/944.delete-columns-to-make-sorted.rs

impl Solution {
    pub fn min_deletion_size(strs: Vec<String>) -> i32 {
        let rows = strs.len();
        let cols = strs.first().unwrap().len();

        // let strs: Vec<Vec<char>> = strs.into_iter().map(|s| s.chars().collect()).collect();

        (0..cols)
            .filter(|&c| {
                for r in 1..rows {
                    if strs[r - 1].as_bytes()[c] > strs[r].as_bytes()[c] {
                        return true;
                    }
                }

                false
            })
            .count() as i32
    }
}